pkg/kotsadmsnapshot/backup.go

```diff
@@ -191,254 +201,512 @@ func CreateApplicationBackup(ctx context.Context, a *apptypes.App, isScheduled b
+			ExcludedNamespaces:      []string{},
+			IncludeClusterResources: ptr.To(true),
+			OrLabelSelectors:        instanceBackupLabelSelectors(metadata.ec != nil),
+			OrderedResources:        map[string]string{},
+			Hooks: velerov1.BackupHooks{
+				Resources: []velerov1.BackupResourceHookSpec{},
+			},
+		},
+	}
+
+	if util.AppNamespace() != metadata.kotsadmNamespace {
+		veleroBackup.Spec.IncludedNamespaces = append(veleroBackup.Spec.IncludedNamespaces, util.AppNamespace())
+	}
+
+	isKurl, err := kurl.IsKurl(k8sClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to check if cluster is kurl")
+	}
+	if isKurl {
+		veleroBackup.Spec.IncludedNamespaces = append(veleroBackup.Spec.IncludedNamespaces, "kurl")
+	}
+
+	isKotsadmClusterScoped := k8sutil.IsKotsadmClusterScoped(ctx, k8sClient, metadata.kotsadmNamespace)
+	if !isKotsadmClusterScoped {
+		// in minimal rbac, a kotsadm role and rolebinding will exist in the velero namespace to give kotsadm access to velero.
+		// we backup and restore those so that restoring to a new cluster won't require that the user provide those permissions again.
+		veleroBackup.Spec.IncludedNamespaces = append(veleroBackup.Spec.IncludedNamespaces, metadata.backupStorageLocationNamespace)
+	}
+
+	for _, appMeta := range metadata.apps {
+		// Don't merge the backup spec if we are using the new improved DR.
+		if !hasAppSpec {
+			err := mergeAppBackupSpec(veleroBackup, appMeta, metadata.kotsadmNamespace, metadata.ec != nil)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to merge app backup spec")
+			}
 		}
+	}
 
-		backupSpec, err := kotsKinds.Marshal("velero.io", "v1", "Backup")
+	veleroBackup.Annotations, err = appendCommonAnnotations(k8sClient, veleroBackup.Annotations, metadata, hasAppSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add annotations to backup")
+	}
+	if hasAppSpec {
+		veleroBackup.Annotations[InstanceBackupTypeAnnotation] = InstanceBackupTypeKotsadm
+	} else {
+		veleroBackup.Annotations[InstanceBackupTypeAnnotation] = InstanceBackupTypeCombined
+	}
+
+	if metadata.ec != nil {
+		veleroBackup.Spec.IncludedNamespaces = append(veleroBackup.Spec.IncludedNamespaces, ecIncludedNamespaces(metadata.ec.installation)...)
+	}
+
+	if metadata.snapshotTTL > 0 {
+		veleroBackup.Spec.TTL = metav1.Duration{
+			Duration: metadata.snapshotTTL,
+		}
+	}
+
+	veleroBackup.Spec.IncludedNamespaces = prepareIncludedNamespaces(veleroBackup.Spec.IncludedNamespaces)
+
+	return veleroBackup, nil
+}
+
+var EnableImprovedDR = false
+
+// getAppInstanceBackup returns a backup spec only if this is Embedded Cluster and the vendor has
+// defined both a backup and restore custom resource (improved DR).
+func getAppInstanceBackupSpec(k8sClient kubernetes.Interface, metadata instanceBackupMetadata) (*velerov1.Backup, error) {
+	// TODO(improveddr): remove this once we have fully implemented the improved DR
+	if !EnableImprovedDR {
+		return nil, nil
+	}
```

Comment:
Is this to be extra cautious or is relying on the absence of `kotsKinds.Backup` and `kotsKinds.Restore` not enough to keep support for the old behaviour?

Reply:
as this is not complete, this disables the feature to allow us to merge the code until we finish it